pkg/virt-handler/virtwrap/manager.go

```diff
@@ -416,7 +536,50 @@ func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) (*api.DomainSpec, error) {
 	return &newSpec, nil
 }
 
+func (l *LibvirtDomainManager) RemoveVMSecrets(vm *v1.VM) error {
+	domName := VMNamespaceKeyFunc(vm)
+
+	// TODO this is horribly inefficient. Maybe there's a better way?
```

Comment:
Out of everything in this patch series, this is the part that has given me the most grief. Somehow we need to garbage collect libvirt secrets associated with VMs once those VMs are undefined.

The problem here is that there's no efficient (or elegant) way of maintaining this relationship between the secret and the vm... at least none that I have found.

The result is I have to index every local secret stored in libvirt and use the secret's 'description' field to maintain this mapping between secret and domain.

I'm not happy with this. I don't have a better solution right now. Maybe someone else has a clever solution?